src/python/zquantum/core/circuit/_gate_test.py

```diff
@@ -1,7 +1,184 @@
 import unittest
 import os
 from ._gate import Gate
+from ._gateset import COMMON_GATES, UNIQUE_GATES
+from ._qubit import Qubit
+from sympy import Symbol
+import qiskit
+from math import pi
+
 
 class TestGate(unittest.TestCase):
-    def test_test(self):
-        pass
\ No newline at end of file
+    def setUp(self):
+        self.two_qubit_gates = ["CNOT", "CZ", "CPHASE", "SWAP"]
+        self.one_parameter_gates = ["PHASE", "Rx", "Ry", "Rz", "CPHASE"]
+
+    def create_gate(self, gate_name, params=None):
+        if gate_name in self.two_qubit_gates:
+            qubit_list = [Qubit(0), Qubit(1)]
+        else:
+            qubit_list = [Qubit(0)]
+        if params is None:
+            params = []
+            if gate_name in self.one_parameter_gates:
+                params = [1.0]
+        gate = Gate(gate_name, qubits=qubit_list, params=params)
+        return gate, qubit_list
+
+    def create_gate_with_symbolic_params(self, gate_name):
+        if gate_name in self.two_qubit_gates:
+            qubit_list = [Qubit(0), Qubit(1)]
+        else:
+            qubit_list = [Qubit(0)]
+        params = []
+        if gate_name in self.one_parameter_gates:
+            params = [Symbol("theta_0")]
```

Comment:
Do we have any two parameter gates?

Reply:
We do have (e.g.: `ZXZ`), but they're not in `COMMON_GATES` so I left them out for now. But I left the structure, so it's easier to expand the tests when we want to decide to do so.

Reply:
Works for me